xbmc/pvr/PVRGUIActions.cpp

```diff
@@ -1130,20 +1134,33 @@ namespace PVR
     if (item->m_bIsFolder)
       return false;
 
+    std::shared_ptr<CPVRRecording> recording;
     const CPVRChannelPtr channel(CPVRItem(item).GetChannel());
-    if ((channel && CServiceBroker::GetPVRManager().IsPlayingChannel(channel)) ||
-        (channel && channel->HasRecording() && CServiceBroker::GetPVRManager().IsPlayingRecording(channel->GetRecording())))
+    if (channel)
     {
-      CGUIMessage msg(GUI_MSG_FULLSCREEN, 0, CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow());
-      CServiceBroker::GetGUI()->GetWindowManager().SendMessage(msg);
-      return true;
+      bool bSwitchToFullscreen = CServiceBroker::GetPVRManager().IsPlayingChannel(channel);
+
+      if (!bSwitchToFullscreen)
+      {
+        recording = CServiceBroker::GetPVRManager().Recordings()->GetRecordingForEpgTag(channel->GetEPGNow());
+        bSwitchToFullscreen = recording && CServiceBroker::GetPVRManager().IsPlayingRecording(recording);
+      }
+
+      if (bSwitchToFullscreen)
```

Comment:
Can this not only a `else`?

Reply:
No, because `bSwitchToFullscreen` gets recalculated in line 1146 and can be false after that.

Reply:
Ah sorry.